### ISSUE TITLE
Fail to upgrade and self-upgrade

### ISSUE THREAD

**Issue by bricewge** [quelpa/quelpa#65]
I stumble upon this issue when I was switching my `.emacs.d` from `el-get` to `quelpa`. And it's still present when I use only the self-upgrading bootstrap snippet code from `README.md`.
The issue appear after I start emacs for the second time (ie: after the creation of the`~/.emacs.d/quelpa`), the `*Messages*` buffer contain this:
```
Quelpa: failed to checkout melpa git repo: `Search failed: "Fetch URL: \\(.*\\)"'
Fetcher: github
Source: quelpa/quelpa

Quelpa: failed to checkout `quelpa': `Search failed: "Fetch URL: \\(.*\\)"' 
```
`quelpa-upgrade` and `quelpa-self-upgrade` failed in the same way.
To sum up the bootstrap work fine and even install other packages, but after it's not possible to manage them.
I use Emacs 24.4.1 on Archlinux.

**Comment by steckerhalter** [comment]
I'm not yet sure what is going on exactly there. The relevant code bits are:

```cl
      (let ((version (condition-case err
                         (package-build-checkout name config dir)
                       (error (quelpa-message t
                                              "failed to checkout `%s': `%s'"
                                              name
                                              (error-message-string err))
                              nil))))
```

so `package-build-checkout` fails which is part of package-build.el from melpa: https://github.com/milkypostman/melpa/blob/master/package-build.el#L241

can you find out what is passed in as `name` `config` and `dir`?
try to use edebug https://www.gnu.org/software/emacs/manual/html_node/elisp/Using-Edebug.html#Using-Edebug

**Comment by steckerhalter** [comment]
apparently upgrading packages fails, so you could try:

```cl
(quelpa 'company :upgrade t)
```
if you evaluate this expression (and company is not yet installed) it would work, the second time it would fail. is that so?

**Comment by steckerhalter** [comment]
ok, so here is the error coming from (`package-build.el`):

```cl
(defun pb/git-repo (dir)
  "Get the current git repo for DIR."
  (pb/run-process-match
   "Fetch URL: \\(.*\\)" dir "git" "remote" "show" "-n" "origin"))
```

it runs `git remote show -n origin` in `dir` and tries to find the fetch url:

```bash
~/.emacs.d/quelpa/build/package-build$ git remote show -n origin
* remote origin
  Fetch URL: git://github.com/milkypostman/melpa.git
  Push  URL: git://github.com/milkypostman/melpa.git
...
```

so in your case the git command doesn't work or does something unexpected (`Search failed` means that it couldn't find `Fetch URL...` in the git output

**Comment by bricewge** [comment]
Oooh, I see.
``` 
$ git remote show -n origin
* distante origin
  URL de rapatriement : git://github.com/milkypostman/melpa.git
  URL push : git://github.com/milkypostman/melpa.git
  Branche HEAD : (not queried)
  Branches distantes : (statut non demandé)
    master
    nxhtml
    revert-2144-master
  Branche locale configurée pour 'git pull' :
    master fusionne avec la distante master
  Référence locale configurée pour 'git push' (statut non demandé) :
    (correspond) pousse vers (correspond)
```
A dirty fix was to add to my init.el `(setenv "LC_ALL" "C")`. I'll report this to melpa.
Thanks for the quick reply!
